src/lib.rs

```diff
@@ -209,18 +209,26 @@ pub async fn init_redis_connection(
     let manager = RedisConnectionManager::new(format!("redis://{ip_addr}:{port}"))?;
     let pool = Pool::builder().build(manager).await?;
 
-    let mut id = CLIENT_ID.write().await;
-    *id = client_id;
-    let mut redis_pool = REDIS_POOL.write().await;
-    *redis_pool = Some(pool);
+    let mut pool_lock = REDIS_POOL
+        .try_write()
+        .map_err(|e| anyhow!("Redis pool error: {}", e))?;
+    *pool_lock = Some(pool);
+
+    let mut client_id_lock = CLIENT_ID
+        .try_write()
+        .map_err(|e| anyhow!("Redis client id error: {}", e))?;
+    *client_id_lock = client_id;
```

Comment:
```suggestion
    {
        let mut pool_lock = REDIS_POOL
            .try_write()
            .map_err(|e| anyhow!("Redis pool error: {}", e))?;
        *pool_lock = Some(pool);
    }

    {
        let mut client_id_lock = CLIENT_ID
            .try_write()
            .map_err(|e| anyhow!("Redis client id error: {}", e))?;
        *client_id_lock = client_id;
    }
```

I'd suggest above. Reasons are
- to drop the lock as quickly as possible after finishing the write.
- Putting things inside the curly braces is pretty common pattern to intend the drop of RwLockGuard. ( refer to the doc https://docs.rs/tokio/latest/tokio/sync/struct.RwLock.html#examples)

Reply:
Done